application/src/main/java/org/thingsboard/server/service/install/SqlDatabaseUpgradeService.java

```diff
@@ -122,7 +122,19 @@ public void upgradeDatabase(String fromVersion) throws Exception {
                 updateSchema("3.6.4", 3006004, "3.7.0", 3007000, null);
                 break;
             case "3.7.0":
-                updateSchema("3.7.0", 3007000, "3.7.1", 3007001, null);
+                updateSchema("3.7.0", 3007000, "3.7.1", 3007001, connection -> {
+                    try {
+                        connection.createStatement().execute("UPDATE rule_node SET " +
+                                "configuration = CASE " +
+                                "  WHEN (configuration::jsonb ->> 'persistAlarmRulesState')::boolean = false THEN jsonb_set(configuration::jsonb, '{fetchAlarmRulesStateOnStart}', 'false'::jsonb)::varchar " +
```

Comment:
1. We can avoid cast to boolean by comparing it with a string. 
2. Please use || operator instead of jsonb_set.